analyzers/tests/SonarAnalyzer.UnitTest/TestCases/SymbolicExecution/Roslyn/NullPointerDereference.CSharp6.cs

```diff
@@ -62,109 +76,111 @@ public void TryCatch4(object o)
             }
         }
 
-    public void Compliant(List<int> list)
-    {
-      var row = list?.Count;
-      if (row != null)
-      {
-        var type = list.ToArray();
-      }
-    }
-
-    public class A
-    {
-      public bool booleanVal { get; set; }
-    }
+        public void Compliant(List<int> list)
+        {
+            var row = list?.Count;
+            if (row != null)
+            {
+                var type = list.ToArray();  // Noncompliant FP, nullability is inferred from result relation
```

Comment:
Is this a TODO on our list? Consider adding a FIXME

Reply:
It leads to a topic called relationships. Should be addressed in [MMF-2563](https://jira.sonarsource.com/browse/MMF-2563) and I consider it fine for now. It's pretty advanced scenario